libopae/plugins/xfpga/feature_plugins/feature_token_list.c

```diff
@@ -109,17 +112,20 @@ struct _fpga_feature_token *feature_token_add(uint32_t type, uint32_t mmio_num,
 
 	err = pthread_mutex_unlock(&ftoken_lock);
 	if (err) {
-		FPGA_ERR("pthread_mutex_unlock() failed: %S", strerror(err));
-		goto out_free;
+		FPGA_ERR("pthread_mutex_unlock() failed: %s", strerror(err));
+		free(tmp);
```

Comment:
You have plugged tmp into the list of tokens on line 110-111, but here you're freeing tmp and corrupting the list.  Consider an alternative approach.